homeassistant/components/samsungtv/config_flow.py

```diff
@@ -78,15 +77,77 @@ def _get_entry(self):
             data=data,
         )
 
+    async def _async_set_device_unique_id(self, raise_on_progress=True):
+        """Set device unique_id."""
+
+        if self._id:
+            await self.async_set_unique_id(
+                self._id, raise_on_progress=raise_on_progress
+            )
+            self._abort_if_unique_id_configured()
+
+        if not self._device_info:
+            await self._async_get_and_check_device_info()
+
+        if uuid := self._device_info.get("device", {}).get(ATTR_UPNP_UDN.lower()):
+            self._id = uuid
+
+        if not self._id:
+            LOGGER.debug(
+                "Property " "%s" " is missing for host %s",
+                ATTR_UPNP_UDN.lower(),
+                self._host,
+            )
+            return self.async_abort(reason=RESULT_ID_MISSING)
+
+        if self._id.startswith("uuid:"):
+            self._id = self._id[5:]
+
+        await self.async_set_unique_id(self._id, raise_on_progress=raise_on_progress)
+        self._abort_if_unique_id_configured()
```

Comment:
```suggestion
```

Move this out of here and into `async_step_zeroconf` and `async_step_ssdp` do the `updates=`

In `async_step_user` don't do `updates=` since we don't know the host is valid